modules/tinymce/src/core/test/ts/browser/NotificationManagerTest.ts

```diff
@@ -115,6 +121,90 @@ UnitTest.asynctest('browser.tinymce.core.NotificationManagerTest', function (suc
     teardown(editor);
   });
 
+  suite.test('TINY-6528: Notification manager should throw events for notification modification', (editor) => {
+    const testMsg: NotificationSpec = {
+      type: 'warning',
+      text: 'unmodified notification text',
+      icon: 'warning',
+      progressBar: true,
+      timeout: 10,
+      closeButton: true
+    };
+    const notifications = editor.notificationManager.getNotifications();
+
+    // Unmodified notification
+    editor.notificationManager.open(testMsg);
+    LegacyUnit.equal(notifications.length, 1, 'Should add notification');
+    LegacyUnit.deepEqual(notifications[0].settings, testMsg, 'Entire notification should be unmodified');
+    teardown(editor);
+
+    editor.on('BeforeOpenNotification', (event) => {
+      event.notification.type = 'success';
+      event.notification.text = 'Modified notification text';
+      event.notification.icon = 'user';
+      event.notification.progressBar = false;
+      event.notification.timeout = 5;
+      event.notification.closeButton = false;
+    });
+
+    const openEvents = [];
+    editor.on('OpenNotification', (event) => openEvents.push(event));
+
+    editor.notificationManager.open(testMsg);
+    LegacyUnit.equal(openEvents.length, 1, 'Should fire OpenNotification event');
+    LegacyUnit.equal(notifications.length, 1, 'Should add notification');
+
+    // Modified notification
+    const modified = notifications[0].settings;
+    LegacyUnit.equal(modified.text, 'Modified notification text', 'Should have modified text');
+    LegacyUnit.equal(modified.type, 'success', 'Should have modified type');
+    LegacyUnit.equal(modified.icon, 'user', 'Should have modified icon');
+    LegacyUnit.equal(modified.progressBar, false, 'Should have modified progressBar');
+    LegacyUnit.equal(modified.timeout, 5, 'Should have modified timeout');
+    LegacyUnit.equal(modified.closeButton, false, 'Should have modified closeButton');
+    teardown(editor);
+  });
+
+  suite.test('TINY-6528: Notification manager should not fire BeforeOpenNotification when fireEvent is false', (editor) => {
```

Comment:
This should probably test the `service_message` instead, since we don't want the `fireEvents` thing public.